File: linguagem_dart/encapsulamento/academia/aluno.dart
import 'academia.dart';
import 'personal.dart';

class Aluno {
  // Atributos da classe Aluno
  String _nome = "";
  late Academia _academia;

  // Getter e setter de nome
  String get getNome => _nome;

  set setNome(String nome) {
    _nome = nome;
  }

  // Getter e setter de localDeTreino
  Academia get getAcademia => _academia;

  set setAcademia(Academia localDeTreino) {
    _academia = localDeTreino;
  }

  // Construtor da classe Aluno
  Aluno(String nomeDoAluno, Academia localDeTreino) {
    setNome = nomeDoAluno;
    setAcademia = localDeTreino;
  }

  // Método que retorna informações
  String retonarInformacoesAluno(Personal personalDoAluno) {
    return "\nO(a) aluno(a) $getNome treina na ${_academia.getNome} acompanhado(a) pelo(a) personal ${personalDoAluno.getNome}";
  }
}

File: linguagem_java/encapsulamento/academia/Aluno.java
package encapsulamento.academia;

public class Aluno {
    // Atributos da classe Aluno
    private String nome;
    private Academia academia;

    // Construtor da classe Aluno
    public Aluno(String nomeDoAluno, Academia localDeTreino) {
        setNome(nomeDoAluno);
        setAcademia(localDeTreino);
    }

    // Getter e setter de nome
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    // Getter e setter de academia
    public Academia getAcademia() {
        return academia;
    }

    public void setAcademia(Academia academia) {
        this.academia = academia;
    }

    // Método que retorna informações
    public String retornarInformacoesAluno(Personal personalDoAluno) {
        return "\nO aluno " + getNome() + " é acompanhado pelo personal " + personalDoAluno.getNome() + " na academia "
                + academia.getNome();
    }

}
